Strip out the front-end code

helenb opened the pull request:
[Link to Ticket](https://torchbox.monday.com/boards/1192293412/pulses/1346361788)

### Description of Changes Made

- Fix pattern library settings to solve general bugs (https://github.com/torchbox/torchbox.com/pull/3/commits/351ec918b68c6347557aa25f3df6c158e2f566a2)
- Strip out classes and old markup, and remove unused templates. Also do a few individual pattern library fixes (https://github.com/torchbox/torchbox.com/pull/3/commits/7329a4b7cc117e78832f30ddf1652a57915d21b5)
- Strip out old js and css, and add what might be useful from kit: https://github.com/torchbox/torchbox.com/pull/3/commits/7b0b638037cd5a235b9f5ac44f413216c28fedb6
- Add kit tooling: https://github.com/torchbox/torchbox.com/pull/3/commits/7b0b638037cd5a235b9f5ac44f413216c28fedb6

Note that I have checked that each template at least works in the pattern library without errors, but haven't checked the yaml in detail - we should do that as part of the build.

I've kept the old cookie js code as that might be useful if we don't add a third party cookie handler.

### How to Test

### Screenshots

<details>
  <summary>Expand to see more</summary>

</details>

### MR Checklist

- [x] Add a description of your pull request and instructions for the reviewer to verify your work.
- [x] If your pull request is for a specific ticket, link to it in the description.
- [x] Stay on point and keep it small so the merge request can be easily reviewed.
- [ ] Consider adding unit tests, especially for bug fixes. If you don't, tell us why.
- [x] Tests and linting passes.
- [ ] Consider updating documentation. If you don't, tell us why.
- [ ] If relevant, list the environments / browsers in which you tested your changes.

chris-lawton commented on package.json:
```diff
@@ -18,91 +18,57 @@
+    "eslint": "^8.51.0",
+    "eslint-config-torchbox": "^1.1.0",
+    "eslint-webpack-plugin": "^4.0.1",
+    "file-loader": "^5.0.2",
```
`file-loader` was deprecated for Webpack v5 so this can go and there's a change needed in `webpack.config` file to switch to `assset-modules` which i'll comment in the file : )

chris-lawton replied:
https://v5.webpack.js.org/loaders/file-loader/

helenb replied:
Ah that's odd as I thought I had copied the latest version of package.json and webpack config from kit.

helenb replied:
Should now be sorted - I committed your change to webpack and removed file-loader

chris-lawton commented:
I'm getting an error when running `npm run build`:

`ERROR in unable to locate '/Users/chrislawton/Sites/torchbox.com/tbx/static_src/images' glob`

I think it's because there's no `static_src/images` directory

chris-lawton commented on webpack.config.js:
```diff
@@ -85,8 +106,19 @@ const options = {
+                // only looks in the fonts folder so pngs in the images folder won't get put in the fonts folder
+                test: /\.(woff|woff2)$/,
+                include: /fonts/,
+                use: {
+                    loader: 'file-loader',
+                    options: {
+                        name: '[name].[ext]',
+                        outputPath: 'fonts/',
+                        publicPath: '../fonts',
+                    },
+                },
```
This was the update to use `asset-modules` after removing `file-loader` 
```suggestion
                type: 'asset/resource',
```

chris-lawton commented on tbx/project_styleguide/templates/patterns/base.html:
```diff
@@ -21,14 +21,14 @@
         {% endif %}
 
-        {% block meta_tags %}{% endblock %}
+        {% comment %} {% block meta_tags %}{% endblock %} {% endcomment %}
```
Question: Is this temporary?

helenb replied:
oh that may be a mistake from when I was debugging lots of issues in the pattern library, which turned out to be some general settings that needed updating.

helenb replied:
Comments removed, and I spotted the yaml file for `base_page.html` was in the wrong folder, so this is also fixed.

chris-lawton commented on tbx/project_styleguide/templates/patterns/molecules/blog-item/blog-item.html:
```diff
@@ -1,39 +1,25 @@
     {% if item.description %}
-        <p class="blog-item__intro">{{ item.description|richtext }}</p>
-        <span class="u-sr-only">,</span>
+        <p>{{ item.description|richtext }}</p>
```
Suggestion: We might not even use this but it needs to be wrapped in a `div` so not to output an empty `<p>` tag
```suggestion
        <div>{{ item.description|richtext }}</div>
```

chris-lawton commented on tbx/project_styleguide/templates/patterns/molecules/event-item/event-item.html:
```diff
@@ -1,42 +1,25 @@
     {% if item.description %}
-        <p class="blog-item__intro">{{ item.description|richtext }}</p>
-        <span class="u-sr-only">,</span>
+        <p>{{ item.description|richtext }}</p>
```
```suggestion
        <div>{{ item.description|richtext }}</div>
```

chris-lawton commented on tbx/project_styleguide/templates/patterns/molecules/help-block/help-block.html:
```diff
@@ -1,51 +1,34 @@
+{% include "patterns/atoms/section-title/section-title.html" with title=title %}
+<h2>
+    {{ intro|richtext }}
+</h2>
```
I'm curious if this works? Would have thought as it has the `|richtext` filter it would leave the `h2` tag empty. (appreciate this was already like this from the previous build)

helenb replied:
It puts a p tag inside the h2 so it does work, but is admittedly odd - I'll just remove the h2.

chris-lawton replied:
Ah good to know!

chris-lawton commented on tbx/project_styleguide/templates/patterns/molecules/streamfield/blocks/cta.html:
```diff
@@ -1,7 +1,5 @@
-        <a href="{{ link.url }}" class="cta__link button">{{ link.text }}</a>
-    {% endwith %}
-</div>
+<p>{{ value.text|richtext }}</p>
```
```suggestion
<div>{{ value.text|richtext }}</div>
```

chris-lawton commented on tbx/static_src/javascript/components/desktop-sub-menu.test.js:
```diff
@@ -0,0 +1,129 @@
+            <div data-desktop-menu="">
+                <nav aria-label="Primary">
+                    <ul class="primary-nav" data-primary-nav="">
+                        <li class="primary-nav__item primary-nav__item--is-parent" data-has-subnav="">
```
Question: Do we need to strip out all these classes as well?

helenb replied:
ah very good spot

helenb replied:
Actually I think I'll leave it for now, as we don't know what the markup will be. I'll just add a todo comment to update it when the navigation is implemented.

chris-lawton commented on tbx/static_src/javascript/components/mobile-sub-menu.js:
```diff
@@ -0,0 +1,52 @@
+            </li>
+        </ul>
+    </li>
+</ul> */
```
I guess things like the above example snippet could be updated once we've got the new markup in place.

chris-lawton commented:
Thanks @helenb - left some comments but feel free to ignore the one's we can fix during build.

helenb commented:
> I'm getting an error when running `npm run build`:
> 
> `ERROR in unable to locate '/Users/chrislawton/Sites/torchbox.com/tbx/static_src/images' glob`
> 
> I think it's because there's no `static_src/images` directory

This should be sorted now. I removed the test images from kit, but clearly that then means git doesn't sync the empty directory. I've added them back in.

chris-lawton commented:
@helenb pulled the changes and working well. Thanks!
